File: common/src/main/java/org/example/module/redis/RedisKeyManager.java
package org.example.module.redis;

import org.example.infra.RedisConfig;
import org.springframework.stereotype.Component;

@Component
public class RedisKeyManager {
    private final RedisConfig redisConfig;

    public RedisKeyManager(RedisConfig redisConfig) {
        this.redisConfig = redisConfig;
    }

    public String getRoomKey(String roomId) {
        return redisConfig.getRoomKeyPrefix() + roomId;
    }

    public String getMessageKey(String messageId) {
        return redisConfig.getMessageKeyPrefix() + messageId;
    }

    public String getSingChatRoomKey(String toUserId, String fromUserId){
        return getRoomKey(toUserId+":"+fromUserId);
    }

    public String getSessionKey(String sessionId){
        return redisConfig.getSessionKeyPrefix() + sessionId;
    }

    public String getUserKey(String userId){
        return redisConfig.getUserKeyPrefix() + userId;
    }

    public String getServerKey(String serverId){
        return redisConfig.getUserKeyPrefix() + serverId;
    }

} 

File: proxy-kotlin/src/main/kotlin/com/chat/proxykotlin/module/redis/RedisKeyManager.kt
package com.chat.proxykotlin.module.redis

import com.chat.proxykotlin.config.redis.RedisConfig
import org.springframework.stereotype.Component

@Component
class RedisKeyManager (
    private val redisConfig: RedisConfig
        ){

    fun getChatRoomKey(roomId: String):String{
        return "${redisConfig.roomKey}$roomId"
    }

    fun getJoinServerKey(roomId: String) :String{
        return "${redisConfig.joinServierKey}$roomId"
    }

    fun getChatRoomJoinMemberList(roomId: String):String{
        return "${redisConfig.roomKey}$roomId"
    }
}